hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MetadataResolver.java

```diff
@@ -56,59 +60,119 @@ public Mapping resolve(String iMapName) {
             return null;
         }
 
-        boolean hd = container.getMapConfig().getInMemoryFormat() == InMemoryFormat.NATIVE;
-        Metadata metadata = hd ? resolveFromHd(container) : resolveFromHeap(iMapName, context);
+        boolean hd = container.getMapConfig().getInMemoryFormat() == InMemoryFormat.NATIVE
+                && !container.getMapConfig().getTieredStoreConfig().isEnabled();
```

Comment:
maybe a note for future code readers about why we have this tiered-store-enabled check.

Reply:
added